test/extensions/filters/http/wasm/test_data/test_cpp.cc

```diff
@@ -100,6 +101,53 @@ bool TestRootContext::onStart(size_t configuration_size) {
   return true;
 }
 
+bool TestRootContext::onConfigure(size_t) {
+  if (test_ == "property") {
+    {
+      // Many properties are not available in the root context.
+      const std::vector<std::string> properties = {
+        "string_state",
+        "metadata",
+        "request",
+        "response",
+        "connection",
+        "connection_id",
+        "upstream",
+        "source",
+        "destination",
+        "cluster_name",
+        "cluster_metadata",
+        "route_name",
+        "route_metadata",
+      };
+      for (const auto& property : properties) {
+        if (getProperty({property}).has_value()) {
+          logWarn("getProperty should not return a value in the root context");
+        }
+      }
+    }
+    {
+      // Some properties are defined in the root context.
+      std::vector<std::pair<std::string, std::string> > properties = {
+        {"plugin_name", "plugin_name"},
+        {"plugin_vm_id", "vm_id"},
+        {"listener_direction", std::string("\x1\0\0\0\0\0\0\0", 8)}, // INBOUND
```

Comment:
The problem is right here.  The std::string does not include a null terminator.  The net result is that when getProperty adds the +1 to the span length, it is going to overrun the buffer that std::string has.

Reply:
I am testing a possible fix : https://github.com/envoyproxy/envoy-wasm/pull/636

Reply:
Thanks. Pushed a potential fix here. Strangely, bazel test reported "cached" results, meaning the Wasm files are identical with or without the last commit.

Reply:
Didn't work.

Reply:
Yah, perhaps you are right, something seems to be odd the initializer_list.